nautobot/extras/tests/test_jobs.py

```diff
@@ -606,6 +606,32 @@ def test_job_profiling(self):
         self.assertTrue(profiling_result.exists())
         profiling_result.unlink()
 
+    def test_job_singleton(self):
+        module = "singleton"
+        name = "TestSingletonJob"
+
+        job_class, _ = get_job_class_and_model(module, name, "local")
+        self.assertTrue(job_class.is_singleton)
+        cache.set(job_class.singleton_cache_key, 1)
+        failed_job_result = create_job_result_and_run_job(module, name)
+
+        self.assertEqual(
+            failed_job_result.status, JobResultStatusChoices.STATUS_FAILURE, msg="Duplicate singleton job didn't error."
+        )
+
+    def test_job_ignore_singleton(self):
+        module = "singleton"
+        name = "TestSingletonJob"
+
+        job_class, _ = get_job_class_and_model(module, name, "local")
+        self.assertTrue(job_class.is_singleton)
+        cache.set(job_class.singleton_cache_key, 1)
+        failed_job_result = create_job_result_and_run_job(module, name, _ignore_singleton_lock=True)
+
+        self.assertEqual(
+            failed_job_result.status, JobResultStatusChoices.STATUS_SUCCESS, msg="Duplicate singleton job didn't error."
+        )
```

Comment:
Can/should we also assert that the cache lock has been cleared?

Reply:
We had discussed some time earlier in this MR that we shouldn't clear the lock, but just ignore it, when we have the flag set. This is the current behaviour. Do you think we should change this again?

Reply:
Ah sorry, wasn't meaning to be inconsistent :-) Doesn't it clear the lock at the end of the job though in any case?

Reply:
That's of course correct. Will add this.